ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/IndexMappingTool.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.tools;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.logging.log4j.util.Strings;
+import org.opensearch.action.admin.indices.get.GetIndexRequest;
+import org.opensearch.action.admin.indices.get.GetIndexResponse;
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.MappingMetadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.common.spi.tools.Parser;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.ml.common.spi.tools.ToolAnnotation;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static org.opensearch.action.support.clustermanager.ClusterManagerNodeRequest.DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT;
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+@ToolAnnotation(IndexMappingTool.NAME)
+public class IndexMappingTool implements Tool {
+    public static final String NAME = "IndexMappingTool";
+
+    private static final String DEFAULT_DESCRIPTION = "Use this tool to get index mapping information.";
+    @Setter
+    @Getter
+    private String name = IndexMappingTool.NAME;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+    @Getter
+    private String type;
+    @Getter
+    private String version;
```

Comment:
Rather than defining these fields for each tool can we create an abstract tool with type and version defined?

Reply:
Good question @arjunkumargiri .  These were not part of the `Tool` interface when I started this PR, and @jngz-es made a direct commit to this dev branch [here](https://github.com/opensearch-project/ml-commons/commit/90d9b310811c682f55ee5415d53ad0be8f671156#diff-f3f27d4e953486c2798c6edef8641779b3211dabbe5374c5bfe1737b3aab2f25) without any documentation of what they mean, so I added them here and in #1582 just to have something that implements the interface.

I was under the understanding that this dev branch was for rapid iteration (thus the non-PR maintainer commit linked above).  I'd really like to not hold up this PR until there is full documentation for the new committed interface that I'm required to implement.

Reply:
Good suggestion, @arjunkumargiri . As @dbwiddis  mentioned, this dev branch is for rapid development to unblock different teams, I can do refactoring afterwards.